lib/internal/modules/cjs/loader.js

```diff
@@ -309,9 +309,7 @@ function readPackage(requestPath) {
 function readPackageScope(checkPath) {
   const rootSeparatorIndex = checkPath.indexOf(path.sep);
   let separatorIndex;
-  while (
-    (separatorIndex = checkPath.lastIndexOf(path.sep)) > rootSeparatorIndex
-  ) {
+  do {
     checkPath = checkPath.slice(0, separatorIndex);
```

Comment:
Isn't `separatorIndex` undefined in the first iteration?

Reply:
Indeed &mdash; was just thinking out loud so far.